File: 1415.the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/solution.go
func getHappyString(n int, k int) string {
    var res string 
    c := 0
    
    var rec func([]byte)
    rec = func(pre []byte) {
        if c == k {
            return
        }
        if len(pre) == n {
            c++
            if c == k {
                res = string(pre)
            }
            return
        }
        for ch := byte('a'); ch <= byte('c'); ch++ {
            if ch != pre[len(pre) - 1] {
                rec(append(pre, ch))
            }
        }
        
    }
    
    for ch := byte('a'); ch <= byte('c'); ch++ {
        rec([]byte{ch})
    }
    
    return res
}

File: 1415.the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/solution.kt
class Solution {
    fun getHappyString(n: Int, k: Int): String {
        return generate("", n).drop(k - 1).firstOrNull().orEmpty()  
    }

    fun generate(pre: String, n: Int): Sequence<String> {
        return sequence {
            if (pre.length == n) {
                yield(pre)
            } else {
                "abc".forEach { ch ->
                    if (pre.lastOrNull() != ch) {
                        yieldAll(generate(pre + ch, n))
                    }
                }
            }
        }
    }
}